pallets/dispenser/src/lib.rs

```diff
@@ -0,0 +1,390 @@
+
+#[derive(Encode, Decode, TypeInfo, Clone, Debug, PartialEq)]
+pub struct EvmTransactionParams {
+	pub value: u128,
+	pub gas_limit: u64,
+	pub max_fee_per_gas: u128,
+	pub max_priority_fee_per_gas: u128,
+	pub nonce: u64,
+	pub chain_id: u64,
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use sp_runtime::traits::AccountIdConversion;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_signet::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		type Currency: Mutate<Self::AccountId, AssetId = AssetId, Balance = Balance>;
+
+		#[pallet::constant]
+		type MinimumRequestAmount: Get<u128>;
+		#[pallet::constant]
+		type MaxDispenseAmount: Get<u128>;
+		#[pallet::constant]
+		type DispenserFee: Get<u128>;
+
+		#[pallet::constant]
+		type FeeAsset: Get<AssetId>;
+		#[pallet::constant]
+		type FaucetAsset: Get<AssetId>;
+
+		#[pallet::constant]
+		type TreasuryAddress: Get<Self::AccountId>;
+
+		#[pallet::constant]
+		type FaucetAddress: Get<EvmAddress>;
+
+		#[pallet::constant]
+		type VaultPalletId: Get<PalletId>;
+
+		#[pallet::constant]
+		type MinFaucetEthThreshold: Get<u128>;
+
+		type WeightInfo: crate::WeightInfo;
+	}
+
+	/*************************** STORAGE ***************************/
+
+	#[pallet::storage]
+	#[pallet::getter(fn dispenser_config)]
+	pub type DispenserConfig<T> = StorageValue<_, DispenserConfigData, OptionQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn current_faucet_balance_wei)]
+	pub type CurrentFaucetBalanceWei<T> = StorageValue<_, u128, ValueQuery>;
+
+	#[derive(Encode, Decode, TypeInfo, Clone, Debug, PartialEq, MaxEncodedLen)]
+	pub struct DispenserConfigData {
+		pub init: bool,
+		pub paused: bool,
+	}
+
+	#[pallet::storage]
+	pub type UsedRequestIds<T: Config> = StorageMap<_, Blake2_128Concat, [u8; 32], (), OptionQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		Initialized,
+		Paused,
+		Unpaused,
+		FundRequested {
+			request_id: [u8; 32],
+			requester: T::AccountId,
+			to: [u8; 20],
+			amount_wei: u128,
+		},
+		FundSucceeded {
+			request_id: [u8; 32],
+		},
+		FundFailed {
+			request_id: [u8; 32],
+		},
+		FaucetBalanceUpdated {
+			old_balance_wei: u128,
+			new_balance_wei: u128,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		AlreadyInitialized,
+		DuplicateRequest,
+		Serialization,
+		NotFound,
+		InvalidOutput,
+		InvalidSignature,
+		InvalidSigner,
+		InvalidRequestId,
+		Paused,
+		AlreadyPaused,
+		AlreadyUnpaused,
+		AmountTooSmall,
+		AmountTooLarge,
+		InvalidAddress,
+		FaucetBalanceBelowThreshold,
+		NotEnoughFunds,
+		NotEnoughFaucetFunds,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::initialize())]
+		pub fn initialize(origin: OriginFor<T>, balance_wei: u128) -> DispatchResult {
```

Comment:
i think we dont need special initialize function. 

we can consider it initliazed/active/ready when there is some balance in "CurrentFauceBalance". 

Pausing feature might be hhelpful - we can leave that but for this , simple bool storage is sufficient. and by default - it will be true. only false value stored.